src/map/status.c

```diff
@@ -12086,33 +12054,178 @@ int status_get_sc_type(sc_type type) {
+				for ( ; w <= pc->max_level[idx][0]; w++ ) {
+					status->HP_table[idx][w] = min(ave * w, battle_config.max_hp);
+				}
+				for ( w = 1; w <= MAX_LEVEL && status->SP_table[iidx][w]; w++ ) {
+					status->SP_table[idx][w] = status->SP_table[iidx][w];
+					total += status->SP_table[idx][w];
+				}
+				ave = total / (w - 1);
+				for ( ; w <= pc->max_level[idx][0]; w++ ) {
+					status->SP_table[idx][w] = min(ave * w, battle_config.max_sp);
+				}
+			}
+		}
+		if ( (temp = libconfig->setting_get_member(jdb, "InheritHP")) ) {
+			int nidx = 0, iidx;
+			const char *iname;
+			while ( (iname = libconfig->setting_get_string_elem(temp, nidx++)) ) {
+				int iclass, w, ave, total = 0;
+				if ( (iclass = pc->check_job_name(iname)) == -1 ) {
+					ShowWarning("status_read_job_db: '%s' trying to inherit unknown '%s' HP!\n", name, iname);
+					continue;
+				}
+				iidx = pc->class2idx(iclass);
+				for ( w = 1; w <= MAX_LEVEL && status->HP_table[iidx][w]; w++ ) {
+					status->HP_table[idx][w] = status->HP_table[iidx][w];
+					total += status->HP_table[idx][w];
+				}
+				ave = total / (w - 1);
+				for ( ; w <= pc->max_level[idx][0]; w++ ) {
+					status->HP_table[idx][w] = min(ave * w, battle_config.max_hp);
+				}
+			}
+		}
+		if ( (temp = libconfig->setting_get_member(jdb, "InheritSP")) ) {
+			int nidx = 0, iidx, ave, total = 0;
+			const char *iname;
+			while ( (iname = libconfig->setting_get_string_elem(temp, nidx++)) ) {
+				int iclass, w;
+				if ( (iclass = pc->check_job_name(iname)) == -1 ) {
+					ShowWarning("status_read_job_db: '%s' trying to inherit unknown '%s' SP!\n", name, iname);
+					continue;
+				}
+				iidx = pc->class2idx(iclass);
+				for ( w = 1; w <= MAX_LEVEL && status->SP_table[iidx][w]; w++ ) {
+					status->SP_table[idx][w] = status->SP_table[iidx][w];
+					total += status->SP_table[idx][w];
+				}
+				ave = total / (w - 1);
+				for ( ; w <= pc->max_level[idx][0]; w++ ) {
+					status->SP_table[idx][w] = min(ave * w, battle_config.max_sp);
+				}
+			}
+		}
+
+		if ( libconfig->setting_lookup_int(jdb, "Weight", &i32) )
+			status->max_weight_base[idx] = i32;
+		else if ( !status->max_weight_base[idx] )
+			status->max_weight_base[idx] = 20000;
+
+		if ( (temp = libconfig->setting_get_member(jdb, "BaseASPD")) ) {
+			int widx = 0;
+			config_setting_t *wpn = NULL;
+			while ( (wpn = libconfig->setting_get_elem(temp, widx++)) ) {
+				int w, wlen = ARRAYLENGTH(wnames);
+				const char *wname = config_setting_name(wpn);
+
+				ARR_FIND(0, wlen, w, strcmp(wnames[w].name, wname) == 0);
+				if ( w != wlen ) {
+					status->aspd_base[idx][wnames[w].id] = libconfig->setting_get_int(wpn);
+				} else {
+					ShowWarning("status_read_job_db: unknown weapon type '%s'!\n", wname);
+				}
+			}
+		}
+
+		if ( (temp = libconfig->setting_get_member(jdb, "HPTable")) ) {
+			int level = 0, ave, total = 0;
+			config_setting_t *hp = NULL;
+			while ( (hp = libconfig->setting_get_elem(temp, level++)) ) {
```

Comment:
What if there are more entries in the libconfig file than the `MAX_LEVEL`? Considering the size of `HP_table[]` is `MAX_LEVEL+1`... Segmentation fault candidate?